src/docs/Dockerfile

```diff
@@ -0,0 +1,47 @@
+## Base ########################################################################
+# Use a larger node image to do the build for native deps (e.g., gcc, python)
+FROM node:lts as base
+
+# Reduce npm log spam and colour during install within Docker
+ENV NPM_CONFIG_LOGLEVEL=warn
+ENV NPM_CONFIG_COLOR=false
+
+# We'll run the app as the `node` user, so put it in their home directory
+WORKDIR /home/node/app
+# Install pnpm
+RUN npm install -g pnpm
+# Copy the source code over
+COPY --chown=node:node . /home/node/app/
+
+## Development #################################################################
+# Define a development target that installs devDeps and runs in dev mode
+FROM base as development
+WORKDIR /home/node/app
+# Install (not ci) with dependencies, and for Linux vs. Linux Musl (which we use for -alpine)
+RUN pnpm install
+# Switch to the node user vs. root
+USER node
+# Expose port 4631
+EXPOSE 4631
+# Start the app in debug mode so we can attach the debugger
+CMD ["pnpm", "run", "start"]
+
+## Production ##################################################################
+# Also define a production target which doesn't use devDeps
+FROM base as production
+WORKDIR /home/node/app
+# We'll install production only deps
+RUN pnpm install --production
+
+## Deploy ######################################################################
+# Use a stable nginx image
+FROM nginx:stable-alpine as deploy
```

Comment:
This stage should get split in two.  You are copying the `node_modules/` into this image layer, which we need for building, but not for hosting with nginx.  I would suggest:

```dockerfile
FROM base as production
WORKDIR /home/node/app
COPY --chown=node:node --from=development /home/node/app/node_modules /home/node/app/node_modules/
RUN pnpm build

FROM nginx:stable-alpine as deploy
WORKDIR /home/node/app
# Copy what we've built from production
COPY --chown=node:node --from=production /home/node/app/build /var/www/data
```

This will leave the massive `node_modules/` folder behind, and we'll end-up with our static web site + nginx.

Reply:
NOTE: this code is untested, just a sketch.

Reply:
I tried this out and it seemed to have built the container but I don't know how to serve the built content locally. Currently researching how to do this.

Reply:
```
$ cd src/docs
$ docker build -t docusaurus:latest .
$ docker run --rm -p 8003:80 docusaurus:latest
```

Go to localhost:8003

Reply:
![2022-02-16 17_00_30-Window](https://user-images.githubusercontent.com/36822198/154364637-5ed5e96d-2eda-45d1-8bcc-1ac629bc9a2b.png)

How is this all happening?? I am baffled XD

Reply:
Put a `/` on the end of `/var/www/data/` when you do the `COPY`

Reply:
I guess nginx is looking for files in `/usr/share/nginx/html`? I'm not 100% sure what I'm looking at

```
PS C:\Users\lecin\Documents\repo\telescope\src\docs>  docker run --rm -p 8003:80 docusaurus:latest
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/02/16 22:15:39 [notice] 1#1: using the "epoll" event method
2022/02/16 22:15:39 [notice] 1#1: nginx/1.20.2
2022/02/16 22:15:39 [notice] 1#1: built by gcc 10.3.1 20210424 (Alpine 10.3.1_git20210424)
2022/02/16 22:15:39 [notice] 1#1: OS: Linux 5.10.76-linuxkit
2022/02/16 22:15:39 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2022/02/16 22:15:39 [notice] 1#1: start worker processes
2022/02/16 22:15:39 [notice] 1#1: start worker process 32
2022/02/16 22:15:39 [notice] 1#1: start worker process 33
172.17.0.1 - - [16/Feb/2022:22:15:50 +0000] "GET / HTTP/1.1" 200 612 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Safari/537.36" "-"
2022/02/16 22:15:50 [error] 33#33: *1 open() "/usr/share/nginx/html/favicon.ico" failed (2: No such file or directory), client: 172.17.0.1, server: localhost, request: "GET /favicon.ico HTTP/1.1", host: "localhost:8003", referrer: "http://localhost:8003/"
172.17.0.1 - - [16/Feb/2022:22:15:50 +0000] "GET /favicon.ico HTTP/1.1" 404 555 "http://localhost:8003/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Safari/537.36" "-"
172.17.0.1 - - [16/Feb/2022:22:15:52 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Safari/537.36" "-"
```

Reply:
From their docs:

```
FROM nginx
COPY static-html-directory /usr/share/nginx/html
```

Reply:
Yes, `/var/www/data` is wrong, change that to `/usr/share/nginx/html`.  Nice catch.

Reply:
Looks like we got something! 

![2022-02-16 17_27_28-Window](https://user-images.githubusercontent.com/36822198/154368084-4459a82c-ac7d-4498-9d03-d06544f334ff.png)

Reply:
@cindyledev can you send a PR to @TDDR with your work, or otherwise co-ordinate passing this off?

Reply:
The Dockerizing Docusaurus part was completed in #2973 , now we just need to do #2993